corehq/apps/geospatial/static/geospatial/js/models.js

```diff
@@ -641,6 +644,25 @@ hqDefine('geospatial/js/models', [
             updateSelectedSavedPolygonParam();
         };
 
+        self.deleteSelectedPolygonFilter = function () {
+            $.ajax({
+                type: 'DELETE',
+                url: geoJSONUrl,
+                dataType: 'json',
+                data: JSON.stringify({'polygon_id': self.selectedSavedPolygonId()}),
+                contentType: "application/json; charset=utf-8",
+                success: function (ret) {
+                    if (!ret.success) {
+                        return alertUser.alert_user(message, 'danger');
+                    }
+                    self.clearSelectedPolygonFilter();
+                    var message = gettext(ret.message) + " " + gettext("Refreshing Page...");
+                    alertUser.alert_user(message, 'success');
+                    window.location.reload();
```

Comment:
Will the user really see the message before reload happens? Why do we need to reload though? We can't just edit the javascript model?

Reply:
> Will the user really see the message before reload happens?

Good point. I could add a timeout for the reload.

> Why do we need to reload though? We can't just edit the javascript model?

I did went with the said approach however ran into below issue.
The saved areas are originally populated from the initial page data when the report data is available. When we reapply the report filters on the page, the saved areas are repopulated from the initial page thus showing the deleted saved area.

Initially I thought of getting latest data through the ajax call but felt it isn't worth making ajax call for every report load.

Reload although not the best solution solves this problem and is also helpful for the Case Grouping page which requires refresh of the page to reload the cases.

Reply:
> however ran into below issue.

Just discovered the issue also happens when we create a new saved area. It is attached to the view model however the view model loses the value when report filters are applied.
I will think about a better solution to handle this and am also open for suggestions.

Reply:
Instead of relying on `initialPageData` for getting the list of saved areas, what if we rather used a `select2` element? We can easily add a request handler to the element (example [here](https://github.com/dimagi/commcare-hq/blob/f4d8aa3a84edccd4bb6cef3e891087b29efb53f1/corehq/apps/data_interfaces/templates/data_interfaces/explore_case_data.html#L36)). We can therefore retrieve saved polygons only when needed.

Reply:
> the saved areas are repopulated from the initial page

Okay, would it be too hard to change how that works?

> It is attached to the view model however the view model loses the value when report filters are applied

Oh so its a bug. Seems like we could address both concerns together in a separate ticket if its going to take time.

Reply:
I like @zandre-eng 's suggestion.

Reply:
Using select2 would add an additional request to fetch saved polygons over using initial page data? Is that something to consider?

Reply:
I don't have the exact bigger context on how the page currently works, but I personally prefer the additional request if it's going to make handling polygons much easier and better.

Reply:
Another benefit in having it as a separate request is that it makes the data load for the initial request smaller. Especially if there are a lot of saved polygons, this can be useful when the user doesn't want to filter by saved polygons.

Reply:
I like the `select2` suggestion as well and would like to look more on that. 
In the interest of time for this ticket , I agree with Manish's  suggestion for creating a new ticket to make this change. 
I can keep the code changes in the same PR if that works for you guys ?

Reply:
Different PR would be nice considering this is already big enough.